engine/src/EntityManager.cpp

```diff
@@ -0,0 +1,72 @@
+//
+// Created by tdancois on 8/4/16.
+//
+
+#include "EntityManager.h"
+#include "TextureManager.h"
+#include "AresException.h"
+
+EntityManager *EntityManager::instance;
+
+EntityManager::EntityManager() : isPlayerSet(false) {}
+
+EntityManager *EntityManager::getInstance() {
+    if (!instance) instance = new EntityManager();
+    return instance;
+}
+
+void EntityManager::addNewPlayer(const unsigned int &id, const std::string &character_texture) {
+    if (isPlayerSet) throw new EntityManagerException("addNewPlayer", "A player is already set");
+    if (entities.count(id) != 0) throw new EntityManagerException("addNewPlayer", "A player is already set");
```

Comment:
Nop, the second message is wrong, thank you.
I prefer the inline condition/action style when an operation clearly atomic, I won't change that.